scipy/stats/_distn_infrastructure.py

```diff
@@ -2015,7 +2015,7 @@ def _mom1_sc(self, m, *args):
         return integrate.quad(self._mom_integ1, 0, 1, args=(m,)+args)[0]
 
     def _pdf(self, x, *args):
-        return _derivative(self._cdf, x, dx=1e-5, args=args, order=5)
+        return _derivative(self._cdf, x, dx=1e-5, args=args, n=5)
```

Comment:
This is quite interesting. It uses 5-point differences, presumably to "improve accuracy". However, the `dx` step is fixed, and isn't scaled with the magnitude of x. 

See https://github.com/scipy/scipy/blob/main/scipy/optimize/_numdiff.py#L144, https://github.com/scipy/scipy/blob/main/scipy/optimize/_numdiff.py#L91.

I would've thought that choosing fixed steps here would degrade the accuracy of computation. I wonder if we should just use `optimize._numdiff.approx_derivative` here, which intelligently chooses step size, albeit with a 3-point stencil. Three step would presumably run in 60% of the time.

@josef-pkt, what do you think?

Reply:
I think this is probably quite a good suggestion. That code looks a lot more robust and it would mean that after the `misc.derivative` deprecation expires we could just get rid of this code as it would be unused. It isn't exactly the nicest to use to due to the fixed dx as pointed out in #17059 and in its current form the accuracy for odd order derivatives is actually pretty poor due to the round off error in representing the central zero weight.

Reply:
`approx_derivative` is really useful, and would be preferred to `derivative` in the majority of cases. However, it only does `2-point`, `3-point`, `complex` steps i.e. isn't setup to use higher N stencils, and it only calculates the first derivative.